libraries/AP_Mission/AP_Mission_Relative.cpp

```diff
@@ -0,0 +1,312 @@
+    }
+
+    if (_restart_behaviour == Restart_Behaviour::RESTART_NOT_TRANSLATED) {
+        _translation.do_translation = false;
+        return;
+    }
+    else
+    {
+        _rel_options_fixed = _rel_options;
+        _translation.calculated = false;
+        _translation.do_translation = true;
+
+        // memorize the Basepoint (location of switching to AUTO)
+        AP::ahrs().get_position(_basepoint_loc); // _basepoint_loc.alt is absolute in [cm] in every case
+
+        AP_Mission::Mission_Command tmp;
+
+        // check if one of the Waypoints has (terrain_alt == 1) -> no translation allowed
+        uint16_t i=0;
+        while (AP::mission()->get_next_nav_cmd(i, tmp) && (_translation.do_translation) && (tmp.id != MAV_CMD_DO_LAND_START)) { // advance to the next command
+            // check if Waypoint with Location
+            if (!(tmp.content.location.lat == 0 && tmp.content.location.lng == 0)) {
+                switch (tmp.id) { // no translation for TAKEOFF or LAND commands
+                    case MAV_CMD_NAV_TAKEOFF:
+                    case MAV_CMD_NAV_LAND:
+                    case MAV_CMD_NAV_VTOL_TAKEOFF:
+                    case MAV_CMD_NAV_VTOL_LAND:
+                        break;
+                    default:
+                        if (tmp.content.location.terrain_alt == 1) {
+                            _translation.do_translation = false;
+                            gcs().send_text(MAV_SEVERITY_NOTICE, "MR: terrain_alt -> NO MOVE POSSIBLE");
+                            return;
+                        }
+                }
+            }
+            i++;// move on to next command
+        }
+
+        // no translation if the distance to Homepoint is too small
+        tmp.content.location = AP::ahrs().get_home();
+        float tmp_distance = tmp.content.location.get_distance(_basepoint_loc); // in [m]
+        if (tmp_distance < _no_translation_radius) {
+             gcs().send_text(MAV_SEVERITY_NOTICE, "MR: distance to Home: %.1fm -> NO translation", tmp_distance);
+            _translation.do_translation = false;
+            return;
+        }
+
+        // calculate amount of rotation
+        switch (_restart_behaviour) {
+            case Restart_Behaviour::RESTART_ROTATED_NORTH:
+            case Restart_Behaviour::RESTART_ROTATED_FIRST_WP:
+                tmp.content.location = AP::ahrs().get_home();
+                // get direction from Homepoint to Basepoint (North-related)
+                _translation.direction = tmp.content.location.get_bearing_to(_basepoint_loc);   // in centidegrees from 0 36000
+                break;
+            case Restart_Behaviour::RESTART_ROTATED_HEADING:
+                // rotation via heading at Basepoint
+                _translation.direction = AP::ahrs().yaw_sensor;
+                break;
+            case Restart_Behaviour::RESTART_ROTATED_CHANNEL:
+                // rotation via RC-channel: 1000mys<->-180deg 1500mys<->0deg 2000mys<->+180deg
```

Comment:
you mean microseconds here, right? so please replace "mys" with "us"

Reply:
done